A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js

```diff
@@ -1092,25 +1106,42 @@ class A320_Neo_CDU_MainDisplay extends FMCMainDisplay {
     }
     checkUpdateFlightPhase() {
         const airSpeed = SimVar.GetSimVarValue("AIRSPEED TRUE", "knots");
+        const groundSpeed = SimVar.GetSimVarValue("GPS GROUND SPEED", "knots");
         const flapsHandlePercent = Simplane.getFlapsHandlePercent();
         const leftThrottleDetent = Simplane.getEngineThrottleMode(0);
         const rightThrottleDetent = Simplane.getEngineThrottleMode(1);
         const highestThrottleDetent = (leftThrottleDetent >= rightThrottleDetent) ? leftThrottleDetent : rightThrottleDetent;
 
-        //End preflight when takeoff power is applied and engines are running
-        if (this.currentFlightPhase <= 2) {
-            if ((highestThrottleDetent == ThrottleMode.TOGA || highestThrottleDetent == ThrottleMode.FLEX_MCT) && SimVar.GetSimVarValue("ENG N1 RPM:1", "Percent") > 15 && SimVar.GetSimVarValue("ENG N1 RPM:2", "Percent") > 15) {
+        if (this.currentFlightPhase <= FlightPhase.FLIGHT_PHASE_TAKEOFF) {
+            const isAirborne = !Simplane.getIsGrounded(); // TODO replace with proper flight mode in future
+            const isTogaFlex = highestThrottleDetent === ThrottleMode.TOGA || highestThrottleDetent === ThrottleMode.FLEX_MCT;
+            const flapsSlatsRetracted = (
+                SimVar.GetSimVarValue("TRAILING EDGE FLAPS LEFT ANGLE", "degrees") === 0 &&
+                SimVar.GetSimVarValue("TRAILING EDGE FLAPS RIGHT ANGLE", "degrees") === 0 &&
+                SimVar.GetSimVarValue("LEADING EDGE FLAPS LEFT ANGLE", "degrees") === 0 &&
+                SimVar.GetSimVarValue("LEADING EDGE FLAPS RIGHT ANGLE", "degrees") === 0
+            );
```

Comment:
**question**: Do we have any failure modelling yet? If not, then I'm guessing we could replace this with a single SimVar call for now, only checking a single slat position as I believe flaps are never extended without slats.

Reply:
Thanks for the conern. We chatted briefly about this on Discord, and the outcome was that there's no concern as long as this is not in a loop, especially as the CDU is not refreshed very often compared to some of the other instruments.